app/controllers/analysis_page_finder_controller.rb

```diff
@@ -0,0 +1,22 @@
+class AnalysisPageFinderController < ApplicationController
+  skip_before_action :authenticate_user!
+  include AnalysisPages
+
+  def show
+    urn = params[:urn]
+    analysis_class = params[:analysis_class]
+
+    school = School.find_by(urn: urn)
+    alert_type = AlertType.find_by(class_name: analysis_class)
```

Comment:
`find_by!` these lookups or check the alert_type exists somewher below?

Reply:
Using find_by! now